app/src/main/kotlin/com/wire/android/ui/home/conversations/UsersTypingIndicator.kt

```diff
@@ -0,0 +1,232 @@
+import androidx.compose.foundation.layout.size
+import androidx.compose.foundation.shape.RoundedCornerShape
+import androidx.compose.material.icons.Icons
+import androidx.compose.material.icons.filled.Edit
+import androidx.compose.material.icons.filled.MoreHoriz
+import androidx.compose.material3.Icon
+import androidx.compose.material3.MaterialTheme
+import androidx.compose.material3.Text
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.getValue
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.res.pluralStringResource
+import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.text.style.TextOverflow
+import androidx.compose.ui.unit.dp
+import com.wire.android.R
+import com.wire.android.model.UserAvatarData
+import com.wire.android.ui.common.UserProfileAvatar
+import com.wire.android.ui.common.colorsScheme
+import com.wire.android.ui.common.dimensions
+import com.wire.android.ui.home.conversations.details.participants.model.UIParticipant
+import com.wire.android.ui.home.conversationslist.model.Membership
+import com.wire.android.ui.theme.wireTypography
+import com.wire.android.util.ui.PreviewMultipleThemes
+import com.wire.kalium.logic.data.id.QualifiedID
+
+@Composable
+fun UsersTypingIndicator(
+    usersTyping: List<UIParticipant>,
+) {
+    if (usersTyping.isNotEmpty()) {
+        val rememberTransition =
+            rememberInfiniteTransition(label = stringResource(R.string.animation_label_typing_indicator_horizontal_transition))
+        Row(
+            verticalAlignment = Alignment.CenterVertically,
+            modifier = Modifier
+                .height(dimensions().spacing24x)
+                .background(
+                    color = colorsScheme().surface,
+                    shape = RoundedCornerShape(dimensions().corner14x),
+                )
+        ) {
+            UsersTypingAvatarPreviews(usersTyping)
+            Text(
+                text = pluralStringResource(
+                    R.plurals.typing_indicator_event_message,
+                    usersTyping.size,
+                    usersTyping.first().name,
+                    usersTyping.size - 1
+                ),
+                style = MaterialTheme.wireTypography.label01.copy(color = colorsScheme().secondaryText),
+                maxLines = 1,
+                overflow = TextOverflow.Ellipsis,
+                modifier = Modifier
+                    .weight(weight = 1f, fill = false)
+                    .padding(
+                        top = dimensions().spacing4x,
+                        bottom = dimensions().spacing4x,
+                        end = dimensions().spacing8x,
+                    )
+            )
+            HorizontalBouncingWritingPen(infiniteTransition = rememberTransition)
+        }
+    }
+}
+
+@Suppress("MagicNumber")
+@Composable
+private fun UsersTypingAvatarPreviews(usersTyping: List<UIParticipant>, maxPreviewsDisplay: Int = 3) {
```

Comment:
Can we extract `maxPreviewsDisplay` as a const so it will be easier later to change it